sharding/utils/marshal_test.go

```diff
@@ -84,5 +83,127 @@ func TestSerializeAndDeserializeblob(t *testing.T) {
 		}
 	}
+}
+
+func TestDeserializeSkipEvm(t *testing.T) {
+	data := make([]byte, 64)
+
+	// Set the indicator byte of the second chunk so that the first flag bit (SKIP_EVM) is true and the length bits equal 1
+	data[32] = 0x81
+	rawBlobs, err := Deserialize(data)
+	if err != nil {
+		t.Errorf("Deserialize failed: %v", err)
+	}
+
+	if len(rawBlobs) != 1 {
+		t.Errorf("Length of blobs incorrect: %v", err)
+	}
+
+	if !rawBlobs[0].flags.skipEvmExecution {
+		t.Errorf("SKIP_EVM flag is not true")
+	}
+
+	if len(rawBlobs[0].data) != 32 {
+		t.Errorf("blob size is not 32: %v", len(rawBlobs[0].data))
+	}
+}
+
+func TestDeserializeSkipEvmFalse(t *testing.T) {
+	// create 64 byte array with the isSkipEVM flag turned on
+	data := make([]byte, 64)
+
+	// Set the indicator byte of the second chunk so that no flag is true and the length bits equal 2
+	data[32] = 0x02
+	rawBlobs, err := Deserialize(data)
+	if err != nil {
+		t.Errorf("Deserialize failed: %v", err)
+	}
+
+	if len(rawBlobs) != 1 {
+		t.Errorf("Length of blobs incorrect: %v", err)
+	}
+
+	if rawBlobs[0].flags.skipEvmExecution {
+		t.Errorf("SKIP_EVM flag is true")
+	}
 
+	if len(rawBlobs[0].data) != 33 {
+		t.Errorf("blob size is not 33: %v", len(rawBlobs[0].data))
+	}
+}
+
+func TestSerializeSkipEvm(t *testing.T) {
+	rawBlobs := make([]*RawBlob, 1)
+	rawBlobs[0] = &RawBlob{data: make([]byte, 32)}
+	rawBlobs[0].data[31] = byte(1)
+	rawBlobs[0].flags.skipEvmExecution = true
+
+	data, err := Serialize(rawBlobs)
+	if err != nil {
+		t.Errorf("Serialize failed: %v", err)
+	}
+
+	if len(data) != 64 {
+		t.Errorf("Length of serialized data incorrect. Should be %v but is %v", 64, len(data))
+	}
+
+	if data[0] != 0 {
+		t.Errorf("Indicating byte for first chunk should be %v but is %v", 0, data[0])
+	}
+
+	if data[32] != 0x81 {
+		t.Errorf("Indicating byte for second chunk should be %v but is %v", 0x81, data[32])
+	}
+}
+
+func TestSerializeSkipEvmFalse(t *testing.T) {
+	rawBlobs := make([]*RawBlob, 1)
+	rawBlobs[0] = &RawBlob{data: make([]byte, 31)}
+
+	data, err := Serialize(rawBlobs)
+	if err != nil {
+		t.Errorf("Serialize failed: %v", err)
+	}
+
+	if len(data) != 32 {
+		t.Errorf("Length of serialized data incorrect. Should be %v but is %v", 32, len(data))
+	}
+
+	if data[0] != 31 {
+		t.Errorf("Indicating byte for first chunk should be %v but is %v", 31, data[0])
+	}
+}
+
+func TestSerializeTestData(t *testing.T) {
+	rawBlobs := make([]*RawBlob, 1)
+	rawBlobs[0] = &RawBlob{data: make([]byte, 60)}
+	blobData := rawBlobs[0].data
+	for i := 0; i < len(blobData); i++ {
+		blobData[i] = byte(i)
+	}
+
+	data, err := Serialize(rawBlobs)
+	if err != nil {
+		t.Errorf("Serialize failed: %v", err)
+	}
+
+	if len(data) != 64 {
+		t.Errorf("Length of serialized data incorrect. Should be %v but is %v", 32, len(data))
+	}
+
+	if data[32] != 29 {
+		t.Errorf("Indicating byte for second chunk should be %v but is %v", 29, data[0])
```

Comment:
Is this correct, should it be `data[32]`?

Reply:
yep, thanks for catching this!